cmd/test/cmd.go

```diff
@@ -210,30 +215,9 @@ func doRun(ctx context.Context, servicesToTest []string, options *Options, ioCtr
 
 	testServices := tree.Ordered()
 
-	var wasBuilt bool
-
-	// make sure the images used for the tests exist. If they don't build them
-	for _, name := range testServices {
-		imgName := manifest.Test[name].Image
-		getImg := func(manifest *model.Manifest) string { return imgName }
-		svc, err := builder.GetServicesToBuildForImage(ctx, manifest, getImg)
-		if err != nil {
-			return analytics.TestMetadata{}, err
-		}
-		if len(svc) < 1 {
-			continue
-		}
-
-		wasBuilt = true
-
-		oktetoLog.Debugf("building services: '%s' needed for test: '%s'", strings.Join(svc, ","), name)
-		if err := builder.Build(ctx, &types.BuildOptions{
-			EnableStages: true,
-			Manifest:     manifest,
-			CommandArgs:  svc,
-		}); err != nil {
-			return analytics.TestMetadata{}, err
-		}
+	wasBuilt, err := doBuild(ctx, manifest, testServices, builder, ioCtrl)
+	if err != nil {
+		return analytics.TestMetadata{}, fmt.Errorf("failed to build services: %w", err)
```

Comment:
I think users will have an hard time if they get this error, to understand this error message. I would say 'image' rather than 'services'. WDYT?

Currently hitting this error results in:

```
 i  Using andreafalzetti @ <context> as context
 x  Failed to build services: the image '${OKTETO_BUILD_NO_IMAGE}' is not in the build section of the manifest
```

I would change it to something like:
<img width="921" alt="Screenshot 2024-06-03 at 22 55 17" src="https://github.com/okteto/okteto/assets/2318450/09184d0d-938b-4349-ac32-fc793ba66c2a">

```suggestion
		return analytics.TestMetadata{}, fmt.Errorf("okteto test needs to build the images defined but failed: %w", err)
```

which results in:

<img width="1158" alt="Screenshot 2024-06-03 at 23 06 54" src="https://github.com/okteto/okteto/assets/2318450/c2e720ab-f1a4-4107-a597-7d194a291ab3">